### DISCUSSION TITLE
Close Buttons on Tabs or Middle-Click Tab to Close

### DISCUSSION THREAD

**Discussion by goober99** [orbitalquark/textadept#197] [Q&A] [answered]
Is there a way to enable close buttons on tabs or make it so that tabs close on middle-click (like tabs in most web browsers)?

**Comment by orbitalquark** [comment, marked as answer]
You should be able to connect to `events.TAB_CLICKED` (https://orbitalquark.github.io/textadept/api.html#events.TAB_CLICKED)

Something like this:

    events.connect(events.TAB_CLICKED, function(index, button)
      if button ~= 2 then return end
      view:goto_buffer(_BUFFERS[index])
      buffer:close()
    end)

**Comment by goober99** [comment]
That did exactly what I wanted! Thanks.
